sources/imdsclient/src/lib.rs

```diff
@@ -45,10 +52,12 @@ impl ImdsClient {
 
     async fn new_impl(imds_base_uri: String) -> Result<Self> {
         let client = Client::new();
-        let session_token = fetch_token(&client, &imds_base_uri).await?;
+        let retry_token_timeout = Duration::from_secs(300);
```

Comment:
This seems like a long time to wait, and quite a jump from what seemed to be 15 seconds before - is this right?  I would tend to think a host is unhealthy and replace it long before 5 minutes of attempting to contact IMDS.

Reply:
This is tied to our timeout in [wicked.service](https://github.com/bottlerocket-os/bottlerocket/commit/6793cdf1241716fdefc43560ed9ba8136d7cb2f8#diff-6b590f776f0f2cce22e6d222e1455d8f218e3492cd569654b8eaa918dd8906ce), but I've added a comment for clarity.